Feature request: Attach to SVG elements

<!-- Thanks for your interest in the project – we appreciate your ideas! Please bear in mind that maintainers might need to request additional information or context before adding a new feature. They need to think about edge cases and how this feature fits in the codebase, so a feature request might take some time to move forward. -->

**Is your feature request related to a problem? Please describe.**
<!-- A clear and concise description of what the problem is. Ex. I'm always frustrated when [...] -->
I was testing a component that renders an SVG element (e.g. `<rect>`), but when I tried to attach it to another SVG element during mount I noticed that my typescript type checker claimed that `attatchTo` must be either a `HTMLElement` or a `string`. Reading the code I noticed that `mount` will always create a `<div>` wrapper element, even when I ignore my type checker and provide an SVG element to the `attacheTo` option.

Here is an example of a mounted SVG element which goes into an HTML element that is inside another SVG element, resulting in this weird structure.

```js
const svg = document.createElementNS("http://www.w3.org/2000/svg", "svg");
document.body.append(svg);

const wrapper = mount(RectComponent, { attachTo: svg });
// error TS2322: Type 'SVGSVGElement' is not assignable to type 'string | HTMLElement | undefined'.
```

```html
<svg>
  <div data-v-app="">
    <rect x="0" y="0" width="10" height="10" ></rect>
  </div>
</svg>
```

**Describe the solution you'd like**
<!-- A clear and concise description of what you want to happen. -->
It would be nice if `mount` would be smarter about which element it wraps around the mounted component. If the `attachTo` element is an SVG element, it should perhaps wrap in a `<g>` instead of a `<div>`. Similarly if the `attachTo` element is a MathML element, perhaps it should wrap it in an `<mrow>`.

**Describe alternatives you've considered**
<!-- A clear and concise description of any alternative solutions or features you've considered. -->

We could also allow to configure the wrapper element factory, so instead of just blindly creating a `div` we could accept a function which creates the desired element.

Or we could simply change the typing of the `attachTo` from `HTMLElement | string` to `Element | string` and leave everything else as is. Having this weird HTML/SVG structure is unlikely to yield any issues inside an isolated testing environment.

**Additional context**
<!-- Add any other context or screenshots about the feature request here. -->

--- a/docs/api/index.md
+++ b/docs/api/index.md
@@ -14,7 +14,7 @@ Note that when mocking dates/timers with Vitest, this must be called after
 
 ```ts
 interface MountingOptions<Props, Data = {}> {
-  attachTo?: HTMLElement | string
+  attachTo?: Element | string
   attrs?: Record<string, unknown>
   data?: () => {} extends Data ? any : Data extends object ? Partial<Data> : any
   props?: (RawProps & Props) | ({} extends Props ? null : never)
@@ -74,7 +74,7 @@ Specify the node to mount the component on. This is not available when using `re
 **Signature:**
 
 ```ts
-attachTo?: HTMLElement | string
+attachTo?: Element | string
 ```
 
 **Details:**
@@ -1882,7 +1882,7 @@ Creates a Wrapper that contains the mounted and rendered Vue component to test w
 
 ```ts
 interface MountingOptions<Props, Data = {}> {
-  attachTo?: HTMLElement | string
+  attachTo?: Element | string
   attrs?: Record<string, unknown>
   data?: () => {} extends Data ? any : Data extends object ? Partial<Data> : any
   props?: (RawProps & Props) | ({} extends Props ? null : never)

--- a/docs/fr/api/index.md
+++ b/docs/fr/api/index.md
@@ -12,7 +12,7 @@ Crée un `Wrapper` qui contient le composant Vue monté et rendu pour le test.
 
 ```ts
 interface MountingOptions<Props, Data = {}> {
-  attachTo?: HTMLElement | string
+  attachTo?: Element | string
   attrs?: Record<string, unknown>
   data?: () => {} extends Data ? any : Data extends object ? Partial<Data> : any
   props?: (RawProps & Props) | ({} extends Props ? null : never)
@@ -72,7 +72,7 @@ Spécifie le nœud où monter le composant.
 **Signature&nbsp;:**
 
 ```ts
-attachTo?: HTMLElement | string
+attachTo?: Element | string
 ```
 
 **Utilisation&nbsp;:**
@@ -1870,7 +1870,7 @@ Crée un `Wrapper` qui contient le composant Vue monté et rendu pour le tester 
 
 ```ts
 interface MountingOptions<Props, Data = {}> {
-  attachTo?: HTMLElement | string
+  attachTo?: Element | string
   attrs?: Record<string, unknown>
   data?: () => {} extends Data ? any : Data extends object ? Partial<Data> : any
   props?: (RawProps & Props) | ({} extends Props ? null : never)

--- a/src/types.ts
+++ b/src/types.ts
@@ -91,7 +91,7 @@ export interface MountingOptions<Props, Data = {}>
    * Can be a valid CSS selector, or an Element connected to the document.
    * @see https://test-utils.vuejs.org/api/#attachto
    */
-  attachTo?: HTMLElement | string
+  attachTo?: Element | string
 }
 
 /**

--- a/tests/mountingOptions/attachTo.spec.ts
+++ b/tests/mountingOptions/attachTo.spec.ts
@@ -26,6 +26,25 @@ describe('options.attachTo', () => {
     wrapper.unmount()
     expect(document.getElementById('attach-to')).toBeNull()
   })
+  it('attaches to a provided SVGElement', () => {
+    const svg = document.createElementNS('http://www.w3.org/2000/svg', 'svg')
+    svg.id = 'root'
+    document.body.appendChild(svg)
+    expect(document.getElementById('root')).not.toBeNull()
+    expect(document.getElementById('attach-to')).toBeNull()
+    const wrapper = mount(TestComponent, {
+      attachTo: svg
+    })
+
+    const root = document.getElementById('root')
+    const rendered = document.getElementById('attach-to')!
+    expect(wrapper.vm.$el.parentNode).not.toBeNull()
+    expect(root).not.toBeNull()
+    expect(rendered).not.toBeNull()
+    expect(rendered.outerHTML).toBe(outerHTML)
+    wrapper.unmount()
+    expect(document.getElementById('attach-to')).toBeNull()
+  })
   it('attaches to a provided CSS selector string', () => {
     const div = document.createElement('div')
     div.id = 'root'

